appIcons.js

```diff
@@ -97,7 +99,11 @@ const MyAppIcon = new Lang.Class({
                    'custom-theme-running-dots-border-width'];
 
         keys.forEach(function(key) {
-            this._dtdSettings.connect('changed::' + key, Lang.bind(this, this._toggleDots));
+            this._signalsHandler.add([
+                this._dtdSettings,
+                'changed::' + key,
+                Lang.bind(this, this._toggleDots)
+            ]);
```

Comment:
We are keeping a single settings object, therefore we need the disconnect machinery. 

Just few considerations:
*  For simplicity, what about having a settings object per dock, and just destroy the settings object on dock remove?
* Even more, what about we completely reload the extension when monitor changes, or the multimonitor option is activated?

Would that make the extension code simpler and less bug prone?